### PR TITLE
Use language level 3 in cythonize

### PR DESCRIPTION
**Pull request by okuta**
Cython will change default language level to 3.

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -371,25 +372,25 @@ cdef class ndarray:
         dtype = get_dtype(dtype)
         if dtype == self.dtype:
             if not copy and (
-                    order_char == 'K' or
-                    order_char == 'A' and (self._c_contiguous or
+                    order_char == b'K' or
+                    order_char == b'A' and (self._c_contiguous or
```
Comment (toslunar) [inline cupy/core/core.pyx]: Fix the indent of the following line.

**Comment by kmaehashi** [comment]
Language level 3 enables `from __future__ import division` (https://github.com/cython/cython/issues/2565), so it's better to replace `/` with `//` where it is needed.

e.g., https://github.com/cupy/cupy/blob/a4749389275b9c9ae60081db6882ae3822296a2a/cupy/core/core.pyx#L3783-L3785

**Comment by kmaehashi** [comment]
> replace / with // where it is needed

I checked by grepping all `*.pyx` with ` / `, and it seems that the above code is the only case we need to fix.

**Comment by okuta** [comment]
Thanks, I fixed.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by kmaehashi** [comment]
Hmm, it seems `unicode` must be used for all `str`-typed arguments when calling CuPy from Python 2...
How about explicitly using language level 2 (to be ready for Cython 3.0) for now, and migrate to language level 3 once when we decide to drop support for Py2?

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by toslunar** [comment]
@okuta Could you resolve conflicts?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**